solidity/contracts/adapters/dia-chainlink-adapter/DIAChainlinkAdapter.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.8.7 <0.9.0;
+
+import '@chainlink/contracts/src/v0.8/interfaces/AggregatorV2V3Interface.sol';
+
+interface IDIAOracleV2 {
+  function getValue(string memory key) external view returns (uint128, uint128);
+}
+
+contract DIAChainlinkAdapter is AggregatorV2V3Interface {
+  /// @notice Thrown when trying to query a round that is not the latest one
+  error OnlyLatestRoundIsAvailable();
+  error NotImplemented();
+
+  /// @notice The address of the underlying DIA Oracle
+  address public immutable DIA_ORACLE;
+  uint8 public immutable decimals;
+  string public description;
+
+  /// @notice The round number we'll use to represent the latest round
+  uint80 internal constant LATEST_ROUND = 0;
+
+  /// @notice Magnitude to convert DIA values to Chainlink values
+  uint256 internal immutable _magnitudeConversion;
+
+  bool internal immutable _decimalsGreaterThanOracle;
+
+  constructor(
+    address _diaOracle,
+    uint8 _oracleDecimals,
+    uint8 _decimals,
+    string memory _description
+  ) {
+    DIA_ORACLE = _diaOracle;
+    decimals = _decimals;
+    description = _description;
+    _decimalsGreaterThanOracle = decimals > _oracleDecimals;
+    _magnitudeConversion = 10**(_decimalsGreaterThanOracle ? _decimals - _oracleDecimals : _oracleDecimals - _decimals);
+  }
+
+  function version() external pure returns (uint256) {
+    revert NotImplemented();
```

Comment:
Should we return "4" like in the API3 adapter? Just for consistency. If we do, I think we can get rid of `NotImplemented`